src/source/Ice/IceAgent.c

```diff
@@ -360,23 +336,19 @@ STATUS iceAgentAddRemoteCandidate(PIceAgent pIceAgent, PCHAR pIceCandidateString
     return retStatus;
 }
 
-STATUS iceAgentGatherLocalCandidate(PIceAgent pIceAgent)
+STATUS iceAgentInitHostCandidate(PIceAgent pIceAgent)
 {
     ENTERS();
 
     STATUS retStatus = STATUS_SUCCESS;
     PKvsIpAddress pIpAddress = NULL;
     PIceCandidate pTmpIceCandidate = NULL, pDuplicatedIceCandidate = NULL, pNewIceCandidate = NULL;
-    KvsIpAddress localIpAddresses[MAX_LOCAL_NETWORK_INTERFACE_COUNT];
-    UINT32 localIpAddressesCount = MAX_LOCAL_NETWORK_INTERFACE_COUNT, i;
+    UINT32 i, localCandidateCount = 0;
     PSocketConnection pSocketConnection = NULL;
-    CHK_STATUS(getLocalhostIpAddresses(localIpAddresses,
-                                       &localIpAddressesCount,
-                                       pIceAgent->kvsRtcConfiguration.iceSetInterfaceFilterFunc,
-                                       pIceAgent->kvsRtcConfiguration.filterCustomData));
+    BOOL locked = FALSE;
 
-    for(i = 0; i < localIpAddressesCount; ++i) {
-        pIpAddress = localIpAddresses + i;
+    for(i = 0; i < pIceAgent->localNetworkInterfaceCount; ++i) {
```

Comment:
any host candidate is valid. The valid state is for STUN and TURN candidates because we need to create placeholders (allocating a socket) but they are not immediately available for use.